gui_app.py

```diff
@@ -496,6 +495,33 @@ def run_installer(self, evt=None):
     def reinstall_deps(self, evt=None):
         utils.check_dependencies()
 
+    def set_file_dialog_environment(self):
+        desktop_environment = os.environ.get('XDG_CURRENT_DESKTOP', '').lower()
+    
+        if 'gnome' in desktop_environment:
+            os.environ['TK_SILENCE_DEPRECATION'] = '1'
+            os.environ['GTK_FILE_CHOOSER_ACTION'] = 'open'
+        elif 'kde' in desktop_environment:
+            os.environ['TK_SILENCE_DEPRECATION'] = '1'
+            os.environ['KDE_FULL_SESSION'] = 'true'
```

Comment:
This code is not working. The function is called at line 521 but I still see the tk dialog.

Reply:
The function `fd.askopenfilename()` is a Tk function that opens a Tk dialog, so it has no connection with the system's file chooser. I agree that Tk's filechooser is weird and ugly, but I think we'd need to change to a different toolkit if we want a better filechooser. It's possible, but it would bring it more dependencies in addition to having to rework all the GUI code.

One improvement I found, though, is a hack way to hide hidden folders and files:
```python
try:
    # Dummy option below means the function throws an exception, which keeps the window from opening
    # but allows the variables to be set afterwards.
    self.root.tk.call('tk_getOpenFile', '-dummyoption')
except TclError:
        pass
self.root.tk.call('set', '::tk::dialog::file::showHiddenBtn', '1')
self.root.tk.call('set', '::tk::dialog::file::showHiddenVar', '0')

# Then call the fd.askopenfilename() function...
```